plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaTable.java

```diff
@@ -0,0 +1,180 @@
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkState;
+import static com.google.common.base.Verify.verify;
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static com.google.common.collect.MoreCollectors.onlyElement;
+import static io.trino.plugin.deltalake.transactionlog.DeltaLakeSchemaSupport.getCheckConstraints;
+import static io.trino.plugin.deltalake.transactionlog.DeltaLakeSchemaSupport.getColumnComments;
+import static io.trino.plugin.deltalake.transactionlog.DeltaLakeSchemaSupport.getColumnInvariants;
+import static io.trino.plugin.deltalake.transactionlog.DeltaLakeSchemaSupport.getColumnTypes;
+import static io.trino.plugin.deltalake.transactionlog.DeltaLakeSchemaSupport.getColumnsMetadata;
+import static io.trino.plugin.deltalake.transactionlog.DeltaLakeSchemaSupport.getColumnsNullability;
+import static io.trino.plugin.deltalake.transactionlog.DeltaLakeSchemaSupport.getExactColumnNames;
+import static io.trino.plugin.deltalake.transactionlog.DeltaLakeSchemaSupport.getGeneratedColumnExpressions;
+import static java.util.Objects.requireNonNull;
+
+@NotThreadSafe
+public class DeltaTable
+{
+    private final List<Column> columns;
+    private final List<String> constraints;
+
+    public static DeltaTable of(MetadataEntry metadataEntry, ProtocolEntry protocolEntry)
+    {
+        requireNonNull(metadataEntry, "metadataEntry is null");
+
+        List<String> columnNames = getExactColumnNames(metadataEntry);
+        Map<String, Object> columnTypes = getColumnTypes(metadataEntry);
+        Map<String, Boolean> columnsNullability = getColumnsNullability(metadataEntry);
+        Map<String, String> columnComments = getColumnComments(metadataEntry);
+        Map<String, Map<String, Object>> columnsMetadata = getColumnsMetadata(metadataEntry);
+        Map<String, String> columnGenerations = getGeneratedColumnExpressions(metadataEntry);
+
+        ImmutableList.Builder<Column> columns = ImmutableList.builderWithExpectedSize(columnNames.size());
+        for (String columnName : columnNames) {
+            columns.add(new Column(
+                    columnName,
+                    columnTypes.get(columnName),
+                    columnsNullability.getOrDefault(columnName, true),
+                    Optional.ofNullable(columnComments.get(columnName)),
+                    columnsMetadata.get(columnName),
+                    columnGenerations.get(columnName)));
+        }
+
+        List<String> constraints = ImmutableList.<String>builder()
+                .addAll(getCheckConstraints(metadataEntry, protocolEntry).values())
+                .addAll(getColumnInvariants(metadataEntry, protocolEntry).values()) // The internal logic for column invariants in Delta Lake is same as check constraints
+                .build();
+
+        return new DeltaTable(columns.build(), constraints);
+    }
+
+    public static DeltaTable empty()
```

Comment:
This method leaves the impression that we may have a table without any columns, which is not true. Please remove this method.

Reply:
I wanted to keep `Column` record private. 

Added the following method to the constructor instead. 
```
checkArgument(!columns.isEmpty(), "columns must not be empty");
```